lib/pry-macro.rb

```diff
@@ -1,76 +1,78 @@
       end
+
+      # By using a stack we can potentially have multiple sessions going. Use at your own peril though...
+      _pry_.record_stack << Pry.history.session_line_count
     end
+  end
 
-    create_command 'stop-macro', 'Stops a recording session, and saves a macro' do
-      banner <<-BANNER
-        Usage: stop-macro [-n name] [-d desc]
+  Pry.commands.create_command 'macro-stop' do
+    description 'Stops recording macro.'
 
-        Stops recording a macro, loads the command, and caches it for later saving if
-        desired. If no name is provided, user will be prompted for one. Descriptions 
-        may be provided, but will default to 'no description'
+    def options(opts)
+      opts.banner <<-BANNER
+      Usage: #{command_name}
+
+      Stops recording a macro, loads the command, and caches it for later saving if desired.
       BANNER
+    end
 
-      def setup
-        if !_pry_.instance_variable_defined?(:@record_stack) && _pry_.record_stack.empty?
-          raise 'Cannot stop recording when no recorder is running'
-        end
+    def setup
+      if !_pry_.instance_variable_defined?(:@record_stack) && _pry_.record_stack.empty?
+        raise 'Cannot stop macro when macro is not start.'
+      end
 
-        session_begin = _pry_.record_stack.pop
-        session_end   = Pry.history.session_line_count
+      session_begin = _pry_.record_stack.pop
+      session_end   = Pry.history.session_line_count
 
-        # Get the history between the start and end of the recording session
-        @history = 
-          Pry.history
-             .to_a
-             .last(session_end - session_begin - 1)
-             .reduce(StringIO.new) { |io, item| io.puts(item); io }
-      end
+      # Get the history between the start and end of the recording session
+      session_history = Pry.history.to_a.last(session_end - session_begin)[0..-2].reject {|e| e == 'edit' }
+      @history = session_history.each_with_object(StringIO.new) {|history, io| io.puts(history) }
+    end
 
-      def options(opts)
-        opts.on :n, :name, "Name to use to define the macro",
-                           optional_argument: true, as: String
-        opts.on :d, :desc, "Description of the macro",
-                           optional_argument: true, as: String
-        opts.on :v, :verbose, "Echoes back the macro definition"
-      end
+    def process
+      # Have to have a name to execute this later
+      opts = _pry_.pry_macro_options
 
-      def process
-        # Have to have a name to execute this later
-        name = opts[:n] || ask('Macro Name: ')
-        desc = opts[:d] || 'no description'
+      name = opts[:name]
+      desc = opts[:desc] || 'no description'
 
-        history_lines = @history.string.lines.map { |s| "      #{s}"}.join.chomp.tap { |h|
-          h.sub!(/^ {6}/,'') # First line won't need the spacing
-        }
+      history_lines = @history.string.lines.map { |s| "      #{s}"}.join.chomp.tap { |h|
```

Comment:
Yeah, this might be something for later that we just use squiggly heredocs to make this irrelevant.